contracts/vaults/RibbonThetaVault.sol

```diff
@@ -536,26 +533,57 @@ contract RibbonThetaVault is DSMath, OptionsVaultStorage {
         address newOption = nextOption;
         require(newOption > PLACEHOLDER_ADDR, "!nextOption");
 
+        uint256 pendingAmount = totalPending();
+        uint256 currentSupply = totalSupply();
+        uint256 currentBalance = assetBalance();
+        uint256 roundStartBalance = currentBalance.sub(pendingAmount);
+        uint16 currentRound = round;
+
+        uint256 singleShare = 10**uint256(_decimals);
+
+        uint256 currentPricePerShare =
+            currentSupply > 0
+                ? singleShare.mul(roundStartBalance).div(currentSupply)
+                : singleShare;
+
+        // After closing the short, if the options expire in-the-money
+        // vault pricePerShare would go down because vault's asset balance decreased.
+        // This ensures that the newly-minted shares do not take on the loss.
+        uint256 mintShares =
+            currentSupply > 0
+                ? pendingAmount.mul(currentSupply).div(roundStartBalance)
+                : pendingAmount;
+
+        // Vault holds temporary custody of the newly minted vault shares
+        _mint(address(this), mintShares);
+
+        uint256 newSupply = currentSupply.add(mintShares);
+
+        // We assume that users cannot queued withdrawals in the first round
+        uint256 queuedWithdrawAmount =
+            newSupply > 0
+                ? queuedWithdrawShares.mul(currentBalance).div(newSupply)
```

Comment:
wanted to resurface the issue with queuedWithdrawShares where when they completeWithdrawals they will be getting it at the pps of that round, not the round in which they initiated withdraw.

[our convo](https://discord.com/channels/@me/814424663551377439/852989414787121202)

Reply:
Will add a TODO comment on this